service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java

```diff
@@ -179,6 +183,33 @@ public void insertEvidenceSummaryDocument(GeneratePdfPayload request, String doc
     claimRepository.save(claim);
   }
 
+  @Override
+  public void updateEvidenceSummaryDocument(UUID eFolderId, MasAutomatedClaimPayload payload) {
+    var claim =
+        claimRepository.findByVbmsId(String.valueOf(payload.getClaimDetail().getBenefitClaimId()));
+    if (claim.isEmpty()) {
+      log.warn("Could not find claim with vbmsId. Could not attach eFolder ID.");
+      return;
+    }
+    if (eFolderId == null) {
+      log.warn("eFolder ID was null, could not attach to claim.");
+    }
+    ClaimEntity claimEntity = claim.get();
+    ContentionEntity contentionEntity = findContention(claimEntity, payload.getDiagnosticCode());
+    if (contentionEntity != null) {
+      Optional<EvidenceSummaryDocumentEntity> esd =
+          evidenceSummaryDocumentRepository.findFirstByContentionIdOrderByCreatedAtDesc(
+              contentionEntity.getId());
+      if (esd.isPresent()) {
+        EvidenceSummaryDocumentEntity esdEntity = esd.get();
+        esdEntity.setFolderId(eFolderId);
+        evidenceSummaryDocumentRepository.save(esdEntity);
+      } else {
+        log.warn("Could not find evidence summary document by contentionId.");
```

Comment:
This is an unexpected/undesirable situation, so make it an error so we can investigate.
```suggestion
        log.error("Could not find evidence summary document by contentionId.");
```